actors/market/tests/harness.rs

```diff
@@ -1120,7 +1181,119 @@ pub fn generate_deal_proposal(
+                payment_start = max(s.last_updated_epoch, d.start_epoch);
+            }
+            let duration = max(0, curr_epoch - payment_start);
+            let payment = duration * &d.storage_price_per_epoch;
+            total_payment += payment;
+            payment_remaining += deal_get_payment_remaining(d, curr_epoch).unwrap();
+            client_unlocked += &d.client_collateral;
+            total_slashed += &d.provider_collateral;
+        }
+    }
+
+    let client_before = get_balance(rt, &client_addr);
+    let provider_before = get_balance(rt, &provider_addr);
+
+    // expected updated amounts
+    let updated_client_escrow = &client_before.balance - &total_payment;
+    let updated_provider_escrow = &provider_before.balance + &total_payment - &total_slashed;
+    let updated_client_locked =
+        &client_before.locked - &total_payment - &payment_remaining - &client_unlocked;
+    let updated_provider_locked = &provider_before.locked - &total_slashed;
+
+    terminate_deals(rt, provider_addr, deal_ids);
+
+    let client_acct = get_balance(rt, &client_addr);
+    let provider_acct = get_balance(rt, &provider_addr);
+    assert_eq!(&updated_client_escrow, &client_acct.balance);
+    assert_eq!(&updated_client_locked, &client_acct.locked);
+    assert_eq!(updated_provider_escrow, provider_acct.balance);
+    assert_eq!(updated_provider_locked, provider_acct.locked);
+
+    (total_payment, total_slashed)
+}
+
+pub fn terminate_deals_no_change(
+    rt: &MockRuntime,
+    client_addr: Address,
+    provider_addr: Address,
+    deal_ids: &[DealID],
+) {
+    let st: State = rt.get_state();
+    let epoch_cid = st.deal_ops_by_epoch;
+
+    // fetch current client  escrow balances
+    let client_acct = get_balance(rt, &client_addr);
+    let provider_acct = get_balance(rt, &provider_addr);
+
+    terminate_deals(rt, provider_addr, deal_ids);
+
+    let st: State = rt.get_state();
+    let new_client_acct = get_balance(rt, &client_addr);
+    let new_provider_acct = get_balance(rt, &provider_addr);
+    assert_eq!(epoch_cid, st.deal_ops_by_epoch);
+    assert_eq!(client_acct, new_client_acct);
+    assert_eq!(provider_acct, new_provider_acct);
+}
+
 pub fn terminate_deals(rt: &MockRuntime, miner_addr: Address, deal_ids: &[DealID]) {
+    let curr_epoch = *rt.epoch.borrow();
+    // calculate the expected amount to be slashed for the provider  that it is burnt
+    let mut total_slashed = TokenAmount::zero();
+    for deal_id in deal_ids {
+        let d = find_deal_proposal(rt, *deal_id);
+        if let Some(d) = d {
+            if curr_epoch < d.end_epoch {
+                total_slashed += d.provider_collateral.clone();
+            }
+        }
+    }
+
+    if total_slashed.is_positive() {
+        rt.expect_send_simple(
+            BURNT_FUNDS_ACTOR_ADDR,
+            METHOD_SEND,
+            None,
+            total_slashed,
+            None,
+            ExitCode::OK,
+        );
+    }
```

Comment:
Check the deals are gone from state (proposal, state, pending etc). Perhaps in terminate_deals_raw.